synapseclient/api/table_services.py

```diff
@@ -0,0 +1,71 @@
+"""
+The purpose of this module is to provide any functions that are needed to interact with
+columns in the Synapse REST API.
+"""
+
+import json
+from typing import TYPE_CHECKING, List, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+    from synapseclient.models import Column
+
+
+async def get_columns(
```

Comment:
I tried giving this function the synapse ID for a project it returned an empty list. Perhaps if the entity is not a table this should be an exception?

Reply:
I don't think that the check should exist in this code. If anything it should be a check on the Synapse Rest API.

1) The issue with adding the check in here is that we would need to perform a separate HTTP request in order to verify what the Synapse ID is that you passed along. This adds extra overheard every time someone would like to call this specific function, since it lacks any other context except for what is passed into it.
2) These functions in the `/api/*` are meant for power users wanting to use the API's directly, and us to be able to build the client functionality. We can think of this as a hierarchy like:

![image](https://github.com/user-attachments/assets/045143ea-c064-4a2b-a208-47691f85b9c3)


Thoughts?

Reply:
That's an excellent point. Perhaps we should ask the platform team for this check? In Schematic several endpoints ask for two synapse entities, usually a fileview and something else. A common user error is two switch those, and the resulting error message has not been useful.

Reply:
A better check may be to enforce in this code:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/93c450bbc1d228b978b9bfa701ab64983555d798/synapseclient/api/entity_factory.py#L352-L365

This is a "generic" factory function that gets an entity from Synapse and pushes it into the relevant dataclass in Python (Project, Table, File, ect...). If the `.get()` type of object that you're calling from doesn't match, we could raise an exception here.

ie:

```
my_table = Table(id="ID of a project").get() # Raise an exception
my_table = Table(id="ID of a table").get() # Proceed as expected
```

Reply:
I created https://sagebionetworks.jira.com/browse/SYNPY-1572 for this